### PR TITLE
Bundy PR Monty Hall

### PR DESCRIPTION
**Pull request by bundy10**
## Kata name:

Monty Hall

## Kata description link:

https://github.com/MYOB-Technology/General_Developer/blob/main/katas/kata-monty-hall/kata-monty-hall.md

# Testing Strategy

### Testing approach and integration testing.

I approached this kata in a classical TDD manner.
where I would use real world objects whenever i could however
I did experiment with a mockiest style. Biggest difference I noticed was when I
introduced a bug into the system any test that used the object that contained this bug would fail. However with mock testing
the only test that would fail are the test whose SUT contains the bug. This gave me the impression that being a mockiest tester proved better test isolation.
However when I wrote a mockiest test and ensure the SUT is talking properly to its dependencies. This differs a lot with a classic test where I would only care about the final state of the SUT and not how the state was derived.
the mock tests are therefore more coupled to the implementations of the SUT dependencies.
changing the nature of calls to dependencies will cause the mock tests to break.

what I found the most useful when choosing to use a mock or a classic test was determining if the SUT has very awkward dependencies.

As you can see in the commits after all refactors I ended up just preferring to use regular tests when possible and used mocks when using a test double was necessary. I am not really a fan of stubs and fakes, I don't like 
writing fixture setups as I believe its too much effort and prefer making mocks with every test. However I can see that with Fixture Setups you can reuse them with multiple test. But I can't get over the broad capabilities a mocking framework gives you
over writing your own stubs and fakes. I believe the best tested program is when you verify the state of the SUT and the behaviour I found that using mocks best achieves this.

# Self-scored competency levels

------------------------------------
### Test Doubles

Level 4

In this Monty Hall kata, I have used Used mocks where SUT dependencies were awkward, entirely in the higher level classes.
The Mocks I used were to be able to take control of randomised variables in order to test exactly what i wanted. I also inserted dummies and spies into some mocks
when I needed to fill parameters and verify some data within a process respectfully.

Some mock tests I have implemented were similar to stubs where i would mock part of the interface to test the SUT and verify a process. seen here: 
_mockGameMode.Setup(gameMode => gameMode.PlayerChooseDoor(It.IsAny<List<Door>>()))
.Callback<List<Door>>(doors => doors[0].PlayerPickedDoor()) I mock part of gameMode to giving it a dummy to test if user is prompted to choose a door.

When i am needed to verify if a process is working within a block of production code, I have used mocks similar to fakes to simulate the dependencies how i want and verify if a process has done what its intended.
seen here:
"_mockGameMode.Setup(gameMode => gameMode.PlayerChooseDoor(It.IsAny<List<Door>>()))
.Callback<List<Door>>(doors => doors[0].PlayerPickedDoor());

_mockGameMode.Setup(gameMode => gameMode.PlayerSwitchOrStayDoor(It.IsAny<List<Door>>()))
.Verifiable();"
 I mock the entire gameMode interface to take a dummy to verify if the player is prompted to switch doors or not.

Spies were used when I needed to verify some specific data within a process. This was done by injecting a spy within mocks to simplify the process needed to test and extract the required data needed. for example in this callback method 
".Callback<List<Door>>(doors => isADoorOpened = doors.Any(door => door.IsDoorOpened()))" I am injecting a boolean to check weather a door is opened or not.

### Keeping Things Small

Level 4

all my classes and methods are absolutely single responsibility. I achieved this by making sure to name methods and classes to reveal intent situated at doing just one thing.
this helped me identify and abstract out any code blocks that were not doing exactly what the name of the method/class was revealing

keeping methods and classes small is relatively easy as following the single responsibility rule It is very important that classes and methods are written
in a way that is easily readable and understandable, this task is significantly easier when there is only one goal in mind for the method/class
code blocks should be as small as possible to achieve this, however with justification as long as the method/class absolutely cannot be abstracted anymore and logic written is valid for the method/class responsibility then code blocks can be large.

### Removing Duplication

Level 4

To combat domain duplication in my code, I always made sure to apply domain sense to the most logical places to ensure each and one of my domains has its own responsibility. 
however for domain logic, I used Door objects that was passed around to the different classes that needed it. Therefore it was challenging to justify single responsibility whilst manipulating door objects.
Although this strategy allowed me to manipulate my door objects in a single place which also avoids domain duplication. 
To avoid code duplication, again I would make sure each piece of code written is uniquely completing a task for example If needed a method to create some doors and open one as well another method that does the same thing but instead injects a car into one
I would instead have a method to create the doors and another to either open a door or inject a car.

### Object Composition

Level 4

I did not use any inheritance in this kata, I saw that the disadvantage of child class dependency would most likely cause problems for me in the future. Especially if I were to refactor the base class or another class in the middle of the inheritance tree.
Instead I inject interfaces in dependent class's constructors and I don't allow any class dependency to be injected other than interfaces.
If I am feeling like there are too many dependencies I would rethink my design in terms of class responsibilities.
However I would only use Inheritance when it makes sense on a significant level, such as when I know for sure i will never need to change base class behaviours.
Reason being why i choose interface dependencies is because of Testability, Removing unnecessary coupling, clear contracts between objects and better code isolation.

in terms of Testability:
Mocking dependencies is a life saver for me.

Removing unnecessary coupling: 
To be able to move things around easier.

clear contracts:
use of interfaces will force me to think in terms of interaction between objects.

better code isolation: 
interfaces will remove nasty side effects around your code as changing an interface implementation won't affect other classes since they depend only on the interface abstraction.

### Command Query Separation

Level 4

I understood that it is important to make sure every single method either be a command that performs a task or a query that returns data to the caller and not both.
with single responsibility it was easy to distinguish this. for example most of my methods were to manipulate doors objects and some methods return data such as a bool depending on the data contained within the door objects.

### Revealing Intent

Level 4

Made sure every single name of Variables/Methods/Classes/Tests clearly gives an idea of what the point of that object is and why it exist.
In other words when someone reads the name of one of my declarations they should already have a clear idea of what that declaration is going to do or what its responsible for.
I found this very easy to achieve by following the single responsibility rule.

### CODE REVIEW

**Comment by bundy10** [comment]
Explanation: My Kata repo is very messed up, I made a few too many mistakes on git therefore I have chosen to create this new repo for this PR.

link to my commits: https://github.com/bundy10/Katas/pull/3/commits

sorry for the inconvenience,
Bundy.

File: MontyHallTests/ConsoleGameTests.cs [hunk 1]
```diff
@@ -0,0 +1,162 @@
...
+
+public class ConsoleGameTests
+{
+    private readonly List<Door> _doors;
+    private readonly ConsoleGame _consoleGame = new();
+    private const int CarDoor = 0; 
+
+    public ConsoleGameTests()
+    {
+        //Assert
```
Comment (kedoodle) [inline MontyHallTests/ConsoleGameTests.cs]:
```suggestion
```
This is a test context constructor, no assertions are made here https://xunit.net/docs/shared-context#constructor

File: MontyHallTests/ConsoleGameTests.cs [hunk 2]
```diff
@@ -0,0 +1,162 @@
...
+        _doors = Enumerable.Range(1, 3)
+            .Select(_ => new Door())
+            .ToList();
+        
+        _doors[CarDoor].InjectCarToDoor();
```
Comment (kedoodle) [inline MontyHallTests/ConsoleGameTests.cs]:
```suggestion
        _doors[PrizeDoor].InjectCarToDoor();
```
Does this read nicely? Will comment on this more later :)

File: MontyHallTests/ConsoleGameTests.cs [hunk 3]
```diff
@@ -0,0 +1,162 @@
...
+        var stringReader = new StringReader(doorToPick);
+        Console.SetIn(stringReader);
+        const int expectedExactlyOneDoorToBePicked = 1;
+        
+        //Act
+        _consoleGame.PlayerChooseDoor(_doors);
+        var numOfDoorsPicked = _doors.FindAll(door => door.HasPlayerPicked()).Count;
+        
+        //Assert
+        Assert.Equal(expectedExactlyOneDoorToBePicked, numOfDoorsPicked);
```
Comment (kedoodle) [inline MontyHallTests/ConsoleGameTests.cs:40]:
```suggestion
        
        //Act
        _consoleGame.PlayerChooseDoor(_doors);
        var numOfDoorsPicked = _doors.FindAll(door => door.HasPlayerPicked()).Count;
        
        //Assert
        Assert.Equal(1, numOfDoorsPicked);
```
maybe just me, but less is more here. alternatively, where the expected value is a bit more complex than a simple `1` `int`, perhaps `var expected = 1;` would suffice

File: MontyHallTests/ConsoleGameTests.cs [hunk 4]
```diff
@@ -0,0 +1,162 @@
...
+using MontyHallv2.Random;
+using Moq;
+
+namespace TestProject1.moq;
+
+public class ConsoleGameTests
+{
+    private readonly List<Door> _doors;
+    private readonly ConsoleGame _consoleGame = new();
+    private const int CarDoor = 0;
```
Comment (kedoodle) [inline MontyHallTests/ConsoleGameTests.cs:15]: consider using `[Theory]` tests to test more scenarios and consolidate some of your test cases e.g. true/false cases can be one test https://learn.microsoft.com/en-us/dotnet/core/testing/unit-testing-with-dotnet-test#create-a-test

File: MontyHallTests/ConsoleGameTests.cs [hunk 5]
```diff
@@ -0,0 +1,162 @@
...
+        _consoleGame.PlayerChooseDoor(_doors);
+        _consoleGame.PlayerSwitchOrStayDoor(_doors);
+        var unpickedDoor = _doors[0].HasPlayerPicked();
+
+        //Assert
+        Assert.False(unpickedDoor);
+    }
+    
+    [Fact]
+    public void GivenPlayerSwitchOrStayDoorIsCalled_WhenADoorIsAlreadyPicked_ThenOneOfTheOtherDoorsWillBePicked()
```
Comment (kedoodle) [inline MontyHallTests/ConsoleGameTests.cs:61]: Could probably be more specific here as you've already "fixed" which door is picked by the player and which door has the prize. Why not just outright test that the picked door after switching is the prize door (given that Monty will reveal a goat door)?

File: MontyHallTests/ConsoleGameTests.cs [hunk 6]
```diff
@@ -0,0 +1,162 @@
...
+        Console.SetOut(stringWriter);
+        
+        //Act
+        _consoleGame.PlayerChooseDoor(_doors);
+        _consoleGame.PlayerSwitchOrStayDoor(_doors);
+        
+        //Assert
+        Assert.Contains(ConstantDialogs.InvalidUserSwitchOrStay, stringWriter.ToString());
+    }
+}
```
Comment (kedoodle) [inline MontyHallTests/ConsoleGameTests.cs:161]:
```suggestion
}

```

File: MontyHallTests/GameMasterTests.cs [hunk 7]
```diff
@@ -0,0 +1,55 @@
...
+        
+        //Act
+        var doors = _gameMaster.GetDoorsIncludingCarDoor();
+        
+        
+        var actualNumberOfDoors = doors.Count;
+        
+        
+        //Assert
+        Assert.Equal(expectedNumberOfDoors, actualNumberOfDoors);
```
Comment (kedoodle) [inline MontyHallTests/GameMasterTests.cs:36]:
```suggestion
        //Arrange
        _mockRandom.Setup(num => num.GetNumberBetweenRange(It.IsAny<int>(), It.IsAny<int>())).Returns(1);
        
        //Act
        var doors = _gameMaster.GetDoorsIncludingCarDoor();
        
        //Assert
        Assert.Equal(3, doors.Count);
```
less is more :) or at least define `expectedNumberOfDoors` closer to where it is needed

people reading the code have to remember that `expectedNumberOfDoors` is set on the first line, but don't know how important it is until the final line. this example is a bit contrived with how simple it is, but hopefully it demonstrates a more general reasoning to keep it simple.

File: MontyHallTests/HostTests.cs [hunk 8]
```diff
@@ -0,0 +1,96 @@
...
+
+    [Fact]
+    public void GivenHostOpensADoorIsCalled_WhenDoorsAreGiven_ThenHostWillOpenDoors()
+    {
+        //Arrange
+        _doors = _gameMaster.GetDoorsIncludingCarDoor();
+        
+        //Act
+        _host.HostOpensDoor(_doors);
+        var openedDoor = _doors.Any(door => door.IsDoorOpened());
```
Comment (kedoodle) [inline MontyHallTests/HostTests.cs:34]: how do we know that a door wasn't already open? consider checking before asserting that no doors are open before calling `_host.HostOpensDoor(_doors);`

File: MontyHallTests/HostTests.cs [hunk 9]
```diff
@@ -0,0 +1,96 @@
...
+    [Fact]
+    public void GivenHostOpensADoorIsCalled_ThenHostOpensADoor()
+    {
+        //Arrange
+        _doors = _gameMaster.GetDoorsIncludingCarDoor();
+
+        //Act
+        _host.HostOpensDoor(_doors);
+        var doorsOpened = _doors.FindAll(door => door.IsDoorOpened());
+        var numOfDoorsOpened = doorsOpened.Count;
```
Comment (kedoodle) [inline MontyHallTests/HostTests.cs:50]: iirc can do a count with lambda function. something like `_doors.Count(d => d.IsDoorOpened())`

File: MontyHallTests/HostTests.cs [hunk 10]
```diff
@@ -0,0 +1,96 @@
...
+        //Act
+        _doors[playerPickedDoor].PlayerPickedDoor();
+        _host.HostOpensDoor(_doors);
+        
+        //Assert
+        Assert.True(_doors[expectedDoorToBeOpened].IsDoorOpened());
+    }
+    
+    [Fact]
+    public void GivenHostGameOutcomeIsCalled_WhenTheDoorThatContainsACarAndIsPickedByPlayer_ThenReturnTrue()
```
Comment (kedoodle) [inline MontyHallTests/HostTests.cs:76]: where is `HostGameOutcome` called as the test name implies?

File: MontyHallTests/HostTests.cs [hunk 11]
```diff
@@ -0,0 +1,96 @@
...
+        //Arrange
+        _mockRandom.Setup(num => num.GetNumberBetweenRange(It.IsAny<int>(), It.IsAny<int>())).Returns(2);
+        _doors = _gameMaster.GetDoorsIncludingCarDoor();
+        _doors[2].PlayerPickedDoor();
+        
+        Assert.True(_doors.First(door => door.HasPlayerPicked()).HasCar());
+    }
+    
+    [Fact]
+    public void GivenHostGameOutcomeIsCalled_WhenTheDoorThatContainsACarAndIsNotPickedByPlayer_ThenReturnFalse()
```
Comment (kedoodle) [inline MontyHallTests/HostTests.cs:87]: as above

File: MontyHallTests/PlayerChoiceValidatorTests.cs [hunk 12]
```diff
@@ -0,0 +1,74 @@
...
+        var invalidDoorChoice = "4\n2";
+        var stringReader = new StringReader(invalidDoorChoice);
+        Console.SetIn(stringReader);
+
+        //Act
+        var userDoorSelection = _consoleGame.GetDoorSelectionFromUser();
+        var isUserDoorSelectionValid = PlayerChoiceValidator.ValidateUserDoorSelection(userDoorSelection);
+
+        //Assert
+        Assert.False(isUserDoorSelectionValid);
```
Comment (kedoodle) [inline MontyHallTests/PlayerChoiceValidatorTests.cs:24]: does the validator need to know about the way that the player choice is input? what if we just test the validator in isolation e.g.:
```suggestion
    public void GivenDoorChoice_WhenUserDoorChoiceIsInvalid_ThenValidatorWillReturnFalse()
    {
        //Arrange
        var userDoorSelection = "4";

        //Act
        var isUserDoorSelectionValid = PlayerChoiceValidator.ValidateUserDoorSelection(userDoorSelection);

        //Assert
        Assert.False(isUserDoorSelectionValid);
```

File: MontyHallTests/PlayerChoiceValidatorTests.cs [hunk 13]
```diff
@@ -0,0 +1,74 @@
...
+        var stringReader = new StringReader(yesToSwitch);
+        Console.SetIn(stringReader);
+
+        //Act
+        var userSwitchOrStayChoice = _consoleGame.GetSwitchOrStayChoiceFromUser();
+        var isUserSwitchOrStayChoiceIsValid = PlayerChoiceValidator.ValidateUserSwitchOrStayChoice(userSwitchOrStayChoice);
+
+        //Assert
+        Assert.True(isUserSwitchOrStayChoiceIsValid);
+    }
```
Comment (kedoodle) [inline MontyHallTests/PlayerChoiceValidatorTests.cs:73]: as above, with regards to testing in isolation

File: MontyHallTests/SimulatorGameTests.cs [hunk 14]
```diff
@@ -0,0 +1,128 @@
...
+    [Fact]
+    public void GivenPlayerChooseDoorIsCalled_WhenSimulatorPlayerChoosesDoor_ThenADoorWillHaveBeenPicked()
+    {
+        
+        //Act
+        _simulatorGame.PlayerChooseDoor(_doors);
+        var isADoorPicked = _doors.Any(door => door.HasPlayerPicked());
+
+        //Assert
+        Assert.True(isADoorPicked);
```
Comment (kedoodle) [inline MontyHallTests/SimulatorGameTests.cs:41]: you probably want to test that exactly one door is picked. this test could pass if somehow all doors are picked

File: MontyHallv2/DoorCreation/Door.cs [hunk 15]
```diff
@@ -0,0 +1,42 @@
+namespace MontyHallv2.DoorCreation;
+
+public class Door
+{
+    private bool _opened;
+    private bool _hasCar;
+    private bool _playerPicked;
+
+    public void OpeningDoor()
```
Comment (kedoodle) [inline MontyHallv2/DoorCreation/Door.cs:9]:
```suggestion
    public void Open()
```

reads nicer when called e.g. `_door.Open()` or `_door.OpenDoor()`

File: MontyHallv2/DoorCreation/Door.cs [hunk 16]
```diff
@@ -0,0 +1,42 @@
+namespace MontyHallv2.DoorCreation;
+
+public class Door
+{
+    private bool _opened;
```
Comment (kedoodle) [inline MontyHallv2/DoorCreation/Door.cs:5]:
```suggestion
    private bool _isOpen;
```
nit pick, but if you think about a door object (you're doing OOP, where instances of theses classes represent an object i.e. an actual door in this case), you can open it and close it more than once.

if open the door, then `_opened` should be set to true. pretty simple. but now if i close the door... should `_opened` be true or false? I could argue that the door has been opened before (as in at least once).

again, a contrived example, but less ambiguity for little effort is great!

File: MontyHallv2/DoorCreation/Door.cs [hunk 17]
```diff
@@ -0,0 +1,42 @@
...
+    private bool _hasCar;
+    private bool _playerPicked;
```
Comment (kedoodle) [inline MontyHallv2/DoorCreation/Door.cs:7]: is this a property of the door itself? imo the door could be more "dumb" and not have to know about whether it has been picked or not, leaving the only properties to be "am I a prize door?" and "am I open?"

you could even argue that the prize door and goat doors are two different kinds of objects. would recommend having a read of [Understanding the Four Rules of Simple Design by Corey Haines](https://leanpub.com/4rulesofsimpledesign). he does a better job at explaining the nuance, way better than I ever could

File: MontyHallv2/DoorCreation/Door.cs [hunk 18]
```diff
@@ -0,0 +1,42 @@
...
+
+    public void PlayerPickedDoor()
+    {
+        _playerPicked = true;
+    }
+
+    public void PlayerUnpickDoor()
+    {
+        _playerPicked = false;
+    }
```
Comment (kedoodle) [inline MontyHallv2/DoorCreation/Door.cs:27]: if you're not keen on a big refactor as suggested with the comment on line 7, perhaps consider some renaming of these functions

```suggestion
    public bool IsPicked()
    {
        return _playerPicked;
    }

    public void PickDoor()
    {
        _playerPicked = true;
    }

    public void UnpickDoor()
    {
        _playerPicked = false;
    }
```

File: MontyHallv2/DoorCreation/Door.cs [hunk 19]
```diff
@@ -0,0 +1,42 @@
...
+    public void InjectCarToDoor()
+    {
+        _hasCar = true;
+    }
```
Comment (kedoodle) [inline MontyHallv2/DoorCreation/Door.cs:31]: the prize never moves from door to door. why not set this at door creation time?

such that you could create your doors something like so (sorry, my C# syntax is rusty):
```
var doors = List<Door>()
{
    new Door{ HasCar = True },
    new Door{ HasCar = False },
    new Door{ HasCar = False }
};
```
where the `HasCar` prop can only be set at construction time and only has a getter

you can randomise the prize door by shuffling the list

File: MontyHallv2/DoorCreation/Door.cs [hunk 20]
```diff
@@ -0,0 +1,42 @@
...
+
+    public bool HasCar()
+    {
+        return _hasCar;
+    }
+
+    public bool IsDoorOpened()
```
Comment (kedoodle) [inline MontyHallv2/DoorCreation/Door.cs:38]:
```suggestion
    public bool IsOpen()
```
we're in the door class. reads nicer `_door.IsOpen()` vs `_door.IsDoorOpened()`

File: MontyHallv2/GameModes/ConsoleGame.cs [hunk 21]
```diff
@@ -0,0 +1,60 @@
...
+        while (!PlayerChoiceValidator.ValidateUserSwitchOrStayChoice(_playerSelection))
+        {
+            Dialog.InvalidSwitchOrStayChoiceInputMessage();
+            _playerSelection = GetSwitchOrStayChoiceFromUser();
+        }
+        
+        if (_playerSelection == "y")
+        {
+            strategy.ToSwitchOrStay(doors);
+        }
```
Comment (kedoodle) [inline MontyHallv2/GameModes/ConsoleGame.cs:44]: this conditional seems to be on if the player has elected to switch?

but then the `strategy.ToSwitchOrStay(doors);` makes it sound like there's still a choice to be made. would be easier to read if the function call was something closer to `Switch(doors)`

File: MontyHallv2/GameModes/Simulate1000.cs [hunk 22]
```diff
@@ -0,0 +1,29 @@
...
+            game.PlayGame();
+            if (game.GetOutComeOfGame())
+            {
+                _gamesWon++;
+            }
+
+            
+        }
+        var losses = ((1000 - _gamesWon) / 1000.0) * 100;
+        var gamesWon = (_gamesWon / 1000.0) * 100;
```
Comment (kedoodle) [inline MontyHallv2/GameModes/Simulate1000.cs:25]: these variable names should suggest they are percentages. consider storing them as decimals and formatting to percentage (and rounding to your desired d.p.) when writing to console https://learn.microsoft.com/en-us/dotnet/standard/base-types/standard-numeric-format-strings#percent-format-specifier-p

File: MontyHallv2/Messages/ConstantDialogs.cs [hunk 23]
```diff
@@ -0,0 +1,11 @@
+namespace MontyHallv2.Messages;
+
+public static class ConstantDialogs
+{
+    public const string InvalidUserDoorSelectionInput = "Please enter a number from 1 to 3";
+    public const string InvalidUserSwitchOrStay = "Please enter y or n";
+    public const string WinningCar = "Congratulations you have won the car";
+    public const string LosingCar = "Sorry you have not won the car :(";
+    public const string PlayerSelectADoor = "Please select a Door from 1 to 3";
+    public const string ToSwitchOrStay = "If you Would like to switch your door please enter 'y' or 'n' to stay";
```
Comment (kedoodle) [inline MontyHallv2/Messages/ConstantDialogs.cs:10]:
```suggestion
    public const string ToSwitchOrStay = "If you would like to switch your door please enter 'y', or 'n' to stay";
```
could be clearer as to mapping "y to switch" and "n to stay"

**Review by kedoodle — COMMENTED** [review]
Hi @bundy10, great job on this kata. I don't write a lot of C#, and it read reasonably well to me; a testament to your ability to reveal intent and keep it simple. That said, I have left a few suggestions for areas to consider. There's no right or wrong - you can definitely disagree with my suggestions.

Some are syntactic, some are a bit more involved. I would suggest that you look into different kinds of ways to drive your unit tests - starting with `[Theory]`. There's all other kinds of test fixtures you can use, but they're a bit more complex and probably unnecessary for this kata.

The main sticking point for me understanding the kata was that the `Door` class being a bit too smart - it's a door. Players are presumably human and are capable of opening doors, but the door doesn't need to know about the player - it could well be a robot that opens the door. In a similar vein, perhaps it should be the game that keeps track of where the prize is and which door is selected by the player.

There's a bit of coupling there with the door. Sounds silly, but there's all these references to the `_doors` variable everywhere as it seems everything needs to know about the doors. Perhaps limiting the coupling in general would be best. Another example of coupling was in your unit tests for the validator. The validator definitely doesn't need to know about how the user input is fetched, or even that its testing user input at all. It just needs to know what is and isn't a valid string (or whatever else input is given to it).

File: MontyHallTests/ConsoleGameTests.cs (hunk 1 above)
Comment (bundy10) [reply MontyHallTests/ConsoleGameTests.cs]: understood, thank you.

File: MontyHallTests/ConsoleGameTests.cs (hunk 2 above)
Comment (bundy10) [reply MontyHallTests/ConsoleGameTests.cs]: understood that cardoor can literally mean a physical car door, prizeDoor makes much more sense. thank you
